### PR TITLE
Added questionnaire linked acceptance test

### PR DESCRIPTION
**Pull request by AdeHarrison**
# How to test?
[as part of ](https://github.com/ONSdigital/census-rm-case-processor/pull/34)

# Links
[Trello](https://trello.com/b/E9lArxTH/rm-census)

# Screenshots (if appropriate):

### CODE REVIEW

**Comment by AdeHarrison** [comment]
[RMC-142(A) Log QuestionnaireLinked event (5)](https://trello.com/c/h29h2Ty7/991-rmc-142a-log-questionnairelinked-event-5)

File: acceptance_tests/features/unaddressed.feature [hunk 1]
```diff
@@ -1,8 +1,15 @@
 Feature: Generating UAC/QID pairs for unaddressed letters & questionnaires
+
   Scenario: UAC/QID pair generated for unaddressed
     When an unaddressed message of questionnaire type 01 is sent
     Then a UACUpdated message not linked to a case is emitted to RH and Action Scheduler
 
+  Scenario: Questionnaire linked to unaddressed
+    Given sample file "sample_for_questionnaire_linked.csv" is loaded
+    And messages are emitted to RH and Action Scheduler with [01] qids
+    Then an UACUpdated message of questionnaire type 01 linking a questionnaire to a case is sent
```
Comment (davidmort) [inline acceptance_tests/features/unaddressed.feature]: the acceptance criteria for this story says about logging/storing the questionnaire_linked case event, not the emission of UAC Updated event

**Review by davidmort — CHANGES_REQUESTED** [review]

File: acceptance_tests/features/unaddressed.feature (hunk 1 above)
Comment (AdeHarrison) [reply acceptance_tests/features/unaddressed.feature]: done

File: acceptance_tests/features/steps/unaddressed.py [hunk 2]
```diff
@@ -23,6 +23,33 @@ def send_unaddressed_message(context, questionnaire_type):
...
 
 
+@when("a Questionnaire Linked message is received")
+def check_linked_message_is_received(context):
+    context.linked_case = context.case_created_events[1]['payload']['collectionCase']
+    context.linked_uac = context.uac_created_events[0]['payload']['uac']
+
+    questionnaire_linked_message = {
+        'event': {
+            'type': 'RESPONSE_RECEIVED',
```
Comment (davidmort) [inline acceptance_tests/features/steps/unaddressed.py]: the type here should be  QUESTIONNAIRE_LINKED

File: acceptance_tests/features/steps/unaddressed.py [hunk 3]
```diff
@@ -32,6 +59,15 @@ def check_uac_message_is_received(context):
     assert context.expected_message_received
 
 
+@then("a Questionnaire Linked event is logged")
+def check_questionnaire_linked_message_is_received(context):
+    context.expected_message_received = False
+    start_listening_to_rabbit_queue(Config.RABBITMQ_RH_OUTBOUND_UAC_QUEUE_TEST,
```
Comment (davidmort) [inline acceptance_tests/features/steps/unaddressed.py]: this isn't really checking if an event is being logged, its checking that a UAC_UPDATED event is being emitted

**Review by davidmort — DISMISSED** [review]

File: acceptance_tests/features/steps/unaddressed.py (hunk 2 above)
Comment (AdeHarrison) [reply acceptance_tests/features/steps/unaddressed.py]: done

File: acceptance_tests/features/steps/unaddressed.py (hunk 3 above)
Comment (AdeHarrison) [reply acceptance_tests/features/steps/unaddressed.py]: done

**Review by LukeLoze1 — APPROVED** [review]
Fine if Dave's comment is fixed

**Review by hbrace — DISMISSED** [review]

**Review by hbrace — APPROVED** [review]
Looks good!

**Pull request merged**